File: Example/SHA1Helper/PCViewController.m
//
//  PCViewController.m
//  SHA1Helper
//
//  Created by JJSon on 01/26/2018.
//  Copyright (c) 2018 JJSon. All rights reserved.
//

#import "PCViewController.h"
#import <SHA1Helper/SHA1Helper.h>
@interface PCViewController ()
@property(nonatomic)BOOL needStop;
@property (weak, nonatomic) IBOutlet UILabel *lbResult;
@end

@implementation PCViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
//    NSString * result = [SHA1Helper sha256:@"hello"];
//    NSString * result2 = [SHA1Helper SM3:@"昨天"];
//    NSLog(@"result: %@", result2);
    
    NSString *filePath = [NSString stringWithFormat:@"%@/Documents/Downloads.zip",NSHomeDirectory()];
    
    
   /* [SHA1Helper calculateFileSHA1WithPath:filePath chunkSizeForReadingData:1024*1024 needCancelHandle:^BOOL{
        return self.needStop;
    } progressHandle:^(CGFloat progress) {
        NSLog(@"%f",progress);
//        printf("progress:%f\n",progress);
    } calculateDoneHandle:^(NSString *result) {
        NSLog(@"done:%@",result);
    } errorHandle:^(NSString *errString) {
        NSLog(@"error:%@",errString);
    }];*/
    
    [SHA1Helper calculateFileSM3WithPath:filePath chunkSizeForReadingData:1024*1024 needCancelHandle:^BOOL{
        return self.needStop;
    } progressHandle:^(CGFloat progress) {
        self.lbResult.text = [NSString stringWithFormat:@"%f",progress];
//        NSLog(@"%f",progress);
//        printf("progress:%f\n",progress);
    } calculateDoneHandle:^(NSString *result) {
//        NSLog(@"done:%@",result);
        self.lbResult.text = result;
    } errorHandle:^(NSString *errString) {
        NSLog(@"error:%@",errString);
    }];
     
}
- (IBAction)stop:(id)sender {
    self.needStop = YES;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Example/SHA1Helper/PCViewController.swift
//
//  PCViewController.swift
//  SHA1Helper_Example
//
//  Created by linjj on 2018/1/26.
//  Copyright © 2018年 JJSon. All rights reserved.
//

import UIKit
import SHA1Helper
class PCViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let filePath = NSHomeDirectory() + "/Documents/microsoft_office_2016_installer.pkg"
        SHA1Help().aaa()
//        SHA1Helper.sha1()
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
